src/main/java/com/google/devtools/build/lib/remote/Chunker.java

```diff
@@ -146,7 +191,35 @@ public void seek(long toOffset) throws IOException {
       reset();
     }
     maybeInitialize();
-    ByteStreams.skipFully(data, toOffset - offset);
+    if (compressed && toOffset > 0) {
```

Comment:
Things are getting complicated here. Can we create a subclass `ZstdChunker` instead to handle zstd compression?